### ISSUE TITLE
`SpatialAnchorManager.ProcessLatestFrame` crashes when scene camera is not tagged `MainCamera`

### ISSUE THREAD

**Issue by johannesvollmer** [Azure/azure-spatial-anchors-samples#301]
**Description**

The line 133, `zNear = mainCamera.nearClipPlane` throws an undhandled null reference exception, if the camera in the scene is not tagged `MainCamera` ([see `Camera.main` documentation](https://docs.unity3d.com/ScriptReference/Camera-main.html)). As a consequence, the call to `session.ProcessFrame(xRCameraFrame.nativePtr.GetPlatformPointer());` will not be executed.

By default, creating a brand-new AR Foundation scene with Unity, will not have a camera does not have this tag assigned.

**Steps to reproduce the issue**
1. create a fresh unity project using the unity `AR` template, which contains a default scene with the `ARSession` and `ARCamera` objects
2. add the azure spatial anchors dependency and call `SpatialAnchorManager.StartSessionAsync`
3. run the project on Android and observe the `NullReferenceException` within the `arCameraManager.frameReceived`
4. the scene will work for the AR Foundation behaviours, but the starting azure session will fail

**Expected behavior**

The exception should probably contain a message which hints at how to solve the problem. For example, the exception could read `no main camera found - have you added the 'MainCamera' tag to your camera game object?`.

**Development information**

 - OS: Windows 10
 - Development Platform\Version: Unity 2020.3.16f1 and 2021.1.21f
 - Azure Spatial Anchors SDK version:  `2.10.2`

**AR Device information:**

 - Device: Galaxy A40
 - OS: Android 11

**Additional context**

You can use the following script to initialize azure spatial anchors:
```c#
            Assert.IsNotNull(this.spatialAnchors, "spatial anchors does not exist");
            Assert.IsNull(this.spatialAnchors.Session, "spatial anchors session already exists");
            Assert.IsFalse(this.spatialAnchors.IsSessionStarted, "spatial anchors session already runs");
            
            this.spatialAnchors.Error += (_, eventArgs) => {
                Debug.LogError("On Spatial Anchors Error: " + eventArgs?.ErrorMessage);
            };
            
            this.spatialAnchors.StopSession();
            await this.spatialAnchors.ResetSessionAsync();
            await this.spatialAnchors.CreateSessionAsync();
            
            this.spatialAnchors.Session.LocationProvider = new PlatformLocationProvider {
                Sensors = {
                    WifiEnabled = false,
                    BluetoothEnabled = false,
                    KnownBeaconProximityUuids = new string[]{},
                    GeoLocationEnabled = false
                }
            };
            
            // starting the session might not have completed yet, wait until is started
            // (might depend on ar foundation session tracking)
            while (ARSession.state != ARSessionState.SessionTracking) {
                Debug.Log("Waiting for AR session state to activate...");
                await Task.Delay(300);
            }
            
            await this.spatialAnchors.StartSessionAsync();

            Assert.IsNotNull(this.spatialAnchors.Session, "failed to create session");
            Assert.IsTrue(this.spatialAnchors.IsSessionStarted, "spatial anchors session was not started");
```

**Comment by oafolabi-msft** [comment]
@johannesvollmer , thank you for bringing this to our attention. Can you please paste a link to line 133 as you described above?  That way I will be certain what file you are referring to?

**Comment by johannesvollmer** [comment]
The value is retrieved in line 786
```c#
protected async virtual void Start()
{
     mainCamera = Camera.main; // this assigns null as the new value
     // ...
```

and then used without a check in line 133
```c#
private void ProcessLatestFrame()
{
    // ....

    var cameraParams = new XRCameraParams
    {
        zNear = mainCamera.nearClipPlane, // this line already results in a null reference exception
        zFar = mainCamera.farClipPlane,
        screenWidth = Screen.width,
        screenHeight = Screen.height,
        screenOrientation = Screen.orientation
    };
```

**Comment by johannesvollmer** [comment]
These lines are from the `SpatialAnchorManager.cs` - in the SDK. I didn't know where to put this feedback, so I put this here, to the examples repository. Is there a public git repository for the source code of the azure SDK? Or where can I give feedback to the SDK itself?

**Comment by phamqduc** [comment]
ASA SDK 2.11.0 was released recently that addresses the bug #36615229 mentioned above. Please give SDK 2.11 a try and let us know if you still experience the issue. Thanks.

**Comment by johannesvollmer** [comment]
Thanks :) I'll try it
